src/ratings/components/reviewComponents/individualReviewComps/StarCount.jsx

```diff
@@ -6,12 +6,43 @@ import Star from './Star.jsx';
 
 
 const StarCount = (props) => {
+  const [listArray, setListArray] = useState(['fill']);
+
+  useEffect(() => {
+    var arrayCounter = []
+    for (var i = 0; i < props.starNumber; i++) {
+      arrayCounter.push('fill');
+    }
+    setListArray(arrayCounter);
+  }, [props.starNumber]);
+
+  // const returnStar = (count) => {
+  //   var arrayCounter = []
+  //   for (var i = 0; i < count; i++) {
+  //     arrayCounter.push('fill');
+  //   }
+  // }
+  const starStyle = {
+    display: 'inliine'
+  }
 
   return (
-    <div>
-      {Array(props.starNumber).fill(<Star />)}
+    <div style={starStyle} key={Math.random()}>
+      {listArray.map(ele => {
+      return (
+        <Star style={starStyle} key={Math.random()} giveKey={Math.random()}/>
```

Comment:
The key prop is definitely needed here. Random is not best practice but I'm not sure what should go here so

Reply:
Thanks! Yeah, the issue is that there is no unique id, since each star is just a copy of itself. I think what I can do is make the sort of artificial array count upwards as it's being built, and then access that count as the key inside the <Star /> component. Thanks for the advice let me see if I can fix it.